tests/new_api/test_device.py

```diff
@@ -652,6 +679,11 @@ def test_derivatives_single_expval(
         self, theta, phi, dev, obs, execute_and_derivatives, batch_obs
     ):
         """Test that the jacobian is correct when a tape has a single expectation value"""
+        if isinstance(obs, qml.SparseHamiltonian) and dev.dtype == np.complex64:
+            pytest.skip(
+                reason="The conversion from qml.Hamiltonian to SparseHamiltonian is only possible with np.complex128"
+            )
```

Comment:
This limitation comes from a hard code in Pennylane [ref](https://github.com/PennyLaneAI/pennylane/blob/060bb9a7479dcf6735671d809da19d500c4b7a89/pennylane/ops/qubit/hamiltonian.py#L522)